user_app/bin/menu/sub_menus/add_user_menu.dart

```diff
@@ -0,0 +1,43 @@
+import '../../model/course.dart';
+import '../../model/user_model.dart';
+import '../../utils/config.dart';
+import '../../utils/logger.dart';
+import '../../utils/store.dart';
+import '../../utils/string_values.dart';
+import '../../utils/utils.dart';
+import '../../utils/validations.dart';
+import '../base/entry.dart';
+
+/// Add user menu
+class AddUserMenu extends Entry {
+  AddUserMenu() : super(StringValues.addUserMenuText) {
+    addSingleValueField<String>(StringValues.fullNameSerial,
+        StringValues.fullNamePrompt, (s) => s, Validations.nameValidation);
+    addSingleValueField<String>(StringValues.addressSerial,
+        StringValues.addressPrompt, (s) => s, Validations.addressValidation);
+    addSingleValueField<int>(StringValues.ageSerial, StringValues.agePrompt,
+        Utils.castToInt, Validations.ageValidation);
+    addSingleValueField<int>(
+        StringValues.rollNumberSerial,
+        StringValues.rollNumberPrompt,
+        Utils.castToInt,
+        Validations.rollNumberValidation);
```

Comment:
Kind of complicated it
But okay 
If you are doing this, then might as well instead of repeating function calls, make it a list where you pass a single object that contains these fields inside them

Reply:
> Kind of complicated it But okay If you are doing this, then might as well instead of repeating function calls, make it a list where you pass a single object that contains these fields inside them

Not getting it. How to simplify.  All fields are necessary. The builder pattern is of no use. So, I have just created a list of files then in the constructor, I am adding each field from the field list.

Reply:
I meant you are calling a sequence of addSingleValueField(). 
Instead store the inputs in an object say SingleValueFiledObject. Have a list of it. 
Loop through the list and call the function with list[index] as the parameter

Reply:
For example 

You are calling like 
calculate(1);
calculate(2);
calculate(3);
.
.
calculate(n);


Make it loop through and call calculate(i) for each iteration

Reply:
Okay understood